### PR TITLE
Dino tricks

### PR DESCRIPTION
**Pull request by Atharva-Phatak**
As discussed in #698. Lightly already had a DINO implementation but it was missing some tricks, discussed in #698.
* I have added the tricks as for freezing last layer which stabilzes the model performance.
* Added the arguments for norm_last_layer which was seen to be done by default, similarly added documentation for freeze_last_layer. 

Please let me know if you need more changes. I have made changes to the examples as well.

### CODE REVIEW

File: examples/pytorch/dino.py [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
 class DINO(torch.nn.Module):
-    def __init__(self, backbone, input_dim):
+    def __init__(self, backbone, input_dim, freeze_last_layer, norm_last_layer):
         super().__init__()
         self.student_backbone = backbone
-        self.student_head = DINOProjectionHead(input_dim, 512, 64, 2048)
+        self.student_head = DINOProjectionHead(input_dim, 512, 64, 2048, freeze_last_layer, norm_last_layer)
         self.teacher_backbone = copy.deepcopy(backbone)
-        self.teacher_head = DINOProjectionHead(input_dim, 512, 64, 2048)
+        self.teacher_head = DINOProjectionHead(input_dim, 512, 64, 2048, freeze_last_layer, norm_last_layer)
```
Comment (guarin) [inline examples/pytorch/dino.py:19]: Here we don't have to pass the `freeze_last_layer` because the teacher head is not frozen.

**Review by guarin — COMMENTED** [review]
Hey thanks a lot for the PR, this is awesome!

I added some small comments about the code :) But I think we can merge afterwards.

File: examples/pytorch/dino.py [hunk 2]
```diff
@@ -11,12 +11,12 @@
 from lightly.models.utils import update_momentum
 
 class DINO(torch.nn.Module):
-    def __init__(self, backbone, input_dim):
+    def __init__(self, backbone, input_dim, freeze_last_layer, norm_last_layer):
         super().__init__()
         self.student_backbone = backbone
-        self.student_head = DINOProjectionHead(input_dim, 512, 64, 2048)
+        self.student_head = DINOProjectionHead(input_dim, 512, 64, 2048, freeze_last_layer, norm_last_layer)
```
[the remark this answers is not in the document]
Comment (Atharva-Phatak) [reply examples/pytorch/dino.py:17]: I will resolve this.

File: lightly/models/modules/heads.py [hunk 3]
```diff
@@ -305,10 +314,20 @@ def __init__(
...
-        self.last_layer.weight_g.requires_grad = False
+        #Option to normalize last layer. 
+        if norm_last_layer:
+            self.last_layer.weight_g.requires_grad = False
+        
+    def cancel_last_layer_gradients(self, current_epoch):
+        """Cancel last layer gradients to stabilize the training
+        
+        """
+        if current_epoch >= self.freeze_last_layer:
```
[the remark this answers is not in the document]
Comment (Atharva-Phatak) [reply lightly/models/modules/heads.py:328]: I will make to it to just -1.

File: examples/pytorch/dino.py (hunk 1 above)
Comment (Atharva-Phatak) [reply examples/pytorch/dino.py:19]: Noted thanks !

**Review by guarin — APPROVED** [review]
This looks good :) Thanks a lot for your contribution @Atharva-Phatak!

**Pull request merged**